### PR TITLE
docs(rbac): Document how to mount the RBAC policy CSV file using the Operator [RHIDP-1783]

### PR DESCRIPTION
**Pull request by rm3l**
**IMPORTANT: Do Not Merge - To be merged by Docs Team Only**

**Version(s):**

- 1.4
- 1.3

**Issue:**

https://issues.redhat.com/browse/RHIDP-1783

**Link to docs preview:**

- New Procedure: [Mounting `policy.csv` file using the Developer Hub Operator](https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-499/authorization/#proc-mounting-the-policy-csv-file-using-the-operator_title-authorization)
- Moved to a dedicated proc file: [Mounting `policy.csv` file to the Developer Hub Helm chart](https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-499/authorization/#proc-mounting-the-policy-csv-file-using-helm_title-authorization)

**Reviews:**
- [ ] SME: @ mention assignee
- [ ] QE: @ mention assignee
- [ ] Docs review: @ mention assignee
- [ ] Additional review: @mention assignee (by writer)

**Additional information:**

### CODE REVIEW

**Comment by rm3l** [comment]
/cc @hmanwani-rh 

Please feel free to reassign it to other folks.
It would be great if this could go into 1.3.x..

File: modules/admin/con-rbac-config-permission-policies-external-file.adoc [hunk 1]
```diff
@@ -131,3 +131,83 @@ permission:
 ----
 --
 
+
+== Mounting `policy.csv` file using the {product-short} Operator
```
Comment (themr0c) [inline modules/admin/con-rbac-config-permission-policies-external-file.adoc:135]: This new content is clearly a new procedure. It should go in a new `proc-mounting-the-policy-csv-file-using-the-rhdhoprator.adoc` file.

Comment (rm3l) [reply modules/admin/con-rbac-config-permission-policies-external-file.adoc]: Done - see [`9c11aea` (#499)](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/499/commits/9c11aea9baac31186f8799ddfbfa501b2c2260db).
For consistency, I've done the same thing by moving the Helm chart procedure to a new `proc-mounting-the-policy-csv-file-using-helm.adoc` file - see [`3414e98` (#499)](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/499/commits/3414e98a013d63270e4df8dd39a7e7b7fe0f8443).

File: modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc [hunk 2]
```diff
@@ -0,0 +1,80 @@
+[id='proc-mounting-the-policy-csv-file-using-the-operator_{context}']
+= Mounting `policy.csv` file using the {product-short} Operator
+
+You can add your `policy.csv` file using the {product-short} Operator by creating a `ConfigMap` and mounting it through your Custom Resource.
```
Comment (hmanwani-rh) [inline modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc:4]:
```suggestion
When the {product} is deployed with the Operator, you can add your `policy.csv` file using the {product-short} Operator by creating a `ConfigMap` and mounting it through your Custom Resource (CR).
```

File: modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  rbac:
+    policies-csv-file: ./rbac-policy.csv
+----
+--
+
+. From the *Developer* perspective in the {ocp-short} web console, select the *Topology* view.
+. Click the overflow menu for the {product} instance that you want to use and select *Edit Backstage* to load the YAML view of the {product} instance.
+. In the CR, enter the name of the custom `rbac-policy` ConfigMap as the value for the `spec.application.extraFiles.configMaps` field. For example:
++
+[source, yaml]
```
Comment (hmanwani-rh) [inline modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc:53]:
```suggestion
.Example custom resource
[source, yaml]
```

File: modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        - name: rbac-policy
+    replicas: 1
+    route:
+      enabled: true
+  database:
+    enableLocalDb: true
+----
+. Click *Save*.
+. Navigate back to the *Topology* view and wait for the {product} pod to start.
+. Click the *Open URL* icon to use the {product} platform with the configuration changes.
```
Comment (hmanwani-rh) [inline modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc:80]:
```suggestion
. Click the *Open URL* icon to access the {product} platform with the updated configuration settings.
```

**Review by hmanwani-rh — COMMENTED** [review]
@rm3l  Thank you for creating the PR. I've added a few suggestions below. Please review them and feel free to reach out if you have any questions.

**Review by hmanwani-rh — APPROVED** [review]

File: modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+When the {product} is deployed with the Operator, you can add your `policy.csv` file using the {product-short} Operator by creating a `ConfigMap` and mounting it through your Custom Resource (CR).
+
+.Prerequisites
+
+* You are logged in to your {ocp-short} account using the {ocp-short} web console.
+* {product} is installed and deployed using the Operator.
+* You have added a custom configuration file to {ocp-short}. For more information, see link:{LinkAdminGuide}[Adding a custom configuration file to {ocp-short}].
++
+//For more information about installing the {product} on {ocp-short} using the Operator, see xref:proc-install-rhdh-ocp-operator_{context}[].
+//replace with a link to the installation guide.
```
Comment (gustavolira) [inline modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc:13]: Should this be here?

File: modules/admin/proc-mounting-the-policy-csv-file-using-helm.adoc [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+You can add your `policy.csv` file to the {product-short} Helm Chart by creating a `configMap` and mounting it.
+
+.Prerequisites
+
+* You are logged in to your {ocp-short} account using the {ocp-short} web console.
+* {product} is installed and deployed using Helm Chart.
++
+//For more information about installing the {product} on {ocp-short} using Helm Chart, see xref:proc-install-rhdh-ocp-helm_{context}[].
+//replace with a link to the installation guide.
```
[the remark this answers is not in the document]
Comment (rm3l) [reply modules/admin/proc-mounting-the-policy-csv-file-using-helm.adoc:14]: I just moved what was already present in the original content over to this new file. Not sure why this comment was there, but I chose to keep it as it was.

**Review by gustavolira — APPROVED** [review]

File: modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc (hunk 5 above)
Comment (hmanwani-rh) [reply modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc:13]: We already have a Jira for replacing the links, and another writer is currently addressing all the links in a single PR. Therefore, we can leave them as they are in this PR: https://issues.redhat.com/browse/RHIDP-3377.

File: assemblies/assembly-configuring-authorization-in-rhdh.adoc [hunk 7]
```diff
@@ -15,6 +15,10 @@ include::modules/admin/con-rbac-config-permission-policies-admin.adoc[leveloffse
 
 include::modules/admin/con-rbac-config-permission-policies-external-file.adoc[leveloffset=+3]
 
+include::modules/admin/proc-mounting-the-policy-csv-file-using-helm.adoc[leveloffset=+4]
+
+include::modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc[leveloffset=+4]
+
```
Comment (themr0c) [inline assemblies/assembly-configuring-authorization-in-rhdh.adoc:21]: The operator procedure should be first:

```suggestion

include::modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc[leveloffset=+4]

include::modules/admin/proc-mounting-the-policy-csv-file-using-helm.adoc[leveloffset=+4]

```

File: modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+      configMaps:
+        - name: rbac-policy
+    replicas: 1
+    route:
+      enabled: true
+  database:
+    enableLocalDb: true
+----
+. Click *Save*.
+. Navigate back to the *Topology* view and wait for the {product} pod to start.
```
Comment (themr0c) [inline modules/admin/proc-mounting-the-policy-csv-file-using-the-operator.adoc:80]: For clarity, consider isolating the verification steps:

```suggestion
.Verification

. Navigate back to the *Topology* view and wait for the {product} pod to start.
```

**Comment by rm3l** [comment]
/cherry-pick release-1.3

(Just so we don't forget)

**Review by themr0c — APPROVED** [review]

**Comment by hmanwani-rh** [comment]
@rm3l Could you please manually cherry-pick this PR to the release-1.3 branch?

**Comment by rm3l** [comment]
> @rm3l Could you please manually cherry-pick this PR to the release-1.3 branch?

@hmanwani-rh This seems to have already been cherry-picked into `release-1.3` and merged by @themr0c - see https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/533

**Pull request merged**